### PR TITLE
Disable client operator's noobaa subscription management

### PR DESCRIPTION
**Pull request by bernerhat**
Set a configuration for client operator's configmap to not manage noobaa's subscription while in provider mode

### CODE REVIEW

File: controllers/storagecluster/provider_server.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ const (
 
 	ocsClientConfigMapName = "ocs-client-operator-config"
 	deployCSIKey           = "DEPLOY_CSI"
+	manageNoobaaSub        = "MANAGE_NOOBAA_SUB"
```
Comment (nb-ohad) [inline controllers/storagecluster/provider_server.go:45]: Why do we use all caps + snake case for configmap keys?

Comment (nb-ohad) [inline controllers/storagecluster/provider_server.go:45]: Lets not use short names in keys 
```suggestion
	manageNoobaaSub        = "MANAGE_NOOBAA_SUBSCRIPTION"
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/provider_server.go (hunk 1 above)
Comment (bernerhat) [reply controllers/storagecluster/provider_server.go:45]: was following the same pattern that was used for CSI above, it can be change imo.

**Comment by Nikhil-Ladha** [comment]
Do we want this change in 4.17 branch? If so, please create a bug and backport the fix.
Thanks.

**Pull request merged**